volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -0,0 +1,331 @@
+    @classmethod
+    def unpack_data(self, mft_record: bytes, offset: int, data_type: str) -> bytes:
+        """Helper to unpack values from the raw mft_record
+        
+        Args:
+            mft_record: 1024 bytes starting from header value as returned by layer read
+            offset: how far in to the record to read
+            data_type: what is the data type to unpack
+
+        Returns:
+            bytes: the unpacked data
+        """
+
+        if data_type == 'unsigned long':
+            return unpack('<L', mft_record[offset:offset+4])[0]
+        elif data_type == 'unsigned short':
+            return unpack('<H', mft_record[offset:offset+2])[0]
+        elif data_type == 'unsigned long long':
+            return unpack('<Q', mft_record[offset:offset+8])[0]
+        elif data_type == 'unsigned char':
+            return unpack('<B', mft_record[offset:offset+1])[0]
+        elif data_type == 'int':
+            return unpack('<I', mft_record[offset:offset+4])[0]
+
+    @classmethod
+    def human_date(self, datetime_object: int) -> str:
+        """Converts a windows epoch to a date time string with a fixed format
+        
+        Args:
+            datetime_object: windows epoch time
+
+        Returns:
+            str: strftime of the windows epoch in UTC
+
+        """
+        dtg = conversion.wintime_to_datetime(datetime_object)
+        return dtg.strftime('%Y-%m-%d %H:%M:%S %z')
+
+    @classmethod
+    def parse_mft_record(self, mft_record: bytes) -> Dict:
+        """Takes an MFT Record and attempts to parse, MFT, SI and FN attributes
+        
+        Args:
+            mft_record: 1024 bytes starting from header value as returned by layer read
+
+        Returns:
+            Dict: a Dictionary that contains the Parse MFT Record
+        """
+        # https://github.com/Invoke-IR/ForensicPosters
+
+        flags = self.unpack_data(mft_record, 22, 'unsigned short')
+        file_type = MFT_FLAGS.get(flags, 'Unknown')
+
+        mft_entry = {
+            "signature": mft_record[:4].decode(),
+            "FixupArrayOffset": self.unpack_data(mft_record, 4, 'unsigned short'),
+            "NumFixupEntries": self.unpack_data(mft_record, 6, 'unsigned short'),
+            "LSN": self.unpack_data(mft_record, 8,  'unsigned long long'),
+            "SequenceValue": self.unpack_data(mft_record, 16, 'unsigned short'),
+            "link_count": self.unpack_data(mft_record, 18, 'unsigned short'),
+            "FirstAttrOffset": self.unpack_data(mft_record, 20, 'unsigned short'),
+            "flags": file_type,
+            "record_number": self.unpack_data(mft_record, 44, 'unsigned long'),
+            "attributes": {
+            "SI": {},
+            "FN": []
+        }
+        }
+
+        attr_offset = mft_entry['FirstAttrOffset']
+        # Check at most for 6 entries
+        for i in range(6):
+            # If we attempt to overread the entry continue out
+            if attr_offset > 1000:
```

Comment:
Should this be `0x1000`?  Hard coding values (even safety limits) like this isn't ideal.  If everything is supposed to fit in a single page of memory, then it might be worth asking the layer how big each page is (once, at the start, for efficiency) and then using that...